src/protocols/secure_channel/CASESession.cpp

```diff
@@ -284,30 +279,51 @@ void CASESession::OnResponseTimeout(ExchangeContext * ec)
 
 CHIP_ERROR CASESession::DeriveSecureSession(CryptoContext & session) const
 {
-    size_t saltlen;
-
-    (void) kKDFSEInfo;
-    (void) kKDFSEInfoLength;
+    switch (mState)
+    {
+    case State::kFinished: {
+        // Generate Salt for Encryption keys
+        size_t saltlen = sizeof(mIPK) + kSHA256_Hash_Length;
 
-    VerifyOrReturnError(mCASESessionEstablished, CHIP_ERROR_INCORRECT_STATE);
+        chip::Platform::ScopedMemoryBuffer<uint8_t> msg_salt;
```

Comment:
No need for a scoped memory buffer here. If you make `saltlen` a `constexpr size_t`, you can use a basic stack buffer. It is very small and this avoids one more possible heap buffer needed where stack would easily be OK.